src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,175 @@
      * Given the time and the number of hours to add, return the changed time.
      */
     public LocalTime addHours(LocalTime localTime, Integer hoursToAdd) {
-        return LocalTime.now();
+        return localTime.plusHours(hoursToAdd);
     }
 
     /**
      * Given the time and the number of minutes to add, return the changed time.
      */
     public LocalTime addMinutes(LocalTime localTime, Integer minutesToAdd) {
-        return LocalTime.now();
+        return localTime.plusMinutes(minutesToAdd);
     }
 
     /**
      * Given the time and the number of seconds to add, return the changed time.
      */
     public LocalTime addSeconds(LocalTime localTime, Integer secondsToAdd) {
-        return LocalTime.now();
+        return localTime.plusSeconds(secondsToAdd);
     }
 
     /**
      * Given the date and the number of weeks to add, return the changed date.
      */
     public LocalDate addWeeks(LocalDate localDate, Integer numberOfWeeks) {
-        return LocalDate.now();
+        return localDate.plusWeeks(numberOfWeeks);
     }
 
     /**
      * Given a random `someDate` date, return one of the following Strings:
-     *     - "`someDate` is after `currentDate`"
-     *                  if `someDate` is in the future relating to the `current date`;
-     *     - "`someDate` is before `currentDate`"
-     *                  if `someDate` is in the past relating to the `current date`;
-     *     - "`someDate` is today"
-     *                  if `someDate` is today;
+     * - "`someDate` is after `currentDate`"
+     * if `someDate` is in the future relating to the `current date`;
+     * - "`someDate` is before `currentDate`"
+     * if `someDate` is in the past relating to the `current date`;
+     * - "`someDate` is today"
+     * if `someDate` is today;
      */
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        LocalDate now = LocalDate.now();
+        if (someDate.isAfter(now)) {
+            return someDate + " is after " + now;
+        }
+        if (someDate.isBefore(now)) {
+            return someDate + " is before " + now;
+        }
+        return someDate + " is today";
     }
 
     /**
      * Given a String representation of a date and some timezone,
      * return LocalDateTime in this timezone.
      */
     public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone) {
-        return LocalDateTime.now();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_PATTERN_SPECIFIC_TIME_ZONE);
+        LocalDateTime dateTime = LocalDateTime.parse(dateInString, formatter);
+        ZoneId zoneId = ZoneId.of(zone);
+        ZoneId zoneIdUtc = ZoneId.of(TIMEZONE_UTC);
+        ZonedDateTime zonedDateTimeUtc = dateTime.atZone(zoneIdUtc);
+        ZonedDateTime zonedDateTime = zonedDateTimeUtc.withZoneSameInstant(zoneId);
+        LocalDateTime localDateTime = zonedDateTime.toLocalDateTime();
+        return localDateTime;
```

Comment:
загалом це рішення може виглядати трішки простіше:
```suggestion
        Instant instant = Instant.parse(dateInString);
        ZonedDateTime zonedDateTime = instant.atZone(ZoneId.of(zone));
        return zonedDateTime.toLocalDateTime();
```

Reply:
Instant and Instant.parse() - miracle )))
Спасибо большое!